content/knowledge_base/flare/consensus/avalanche/specification.md

```diff
@@ -0,0 +1,210 @@
+
+Similar to Snowball, sampling in Avalanche will create a positive feedback for the preference of a single transaction in its conflict set.
+When a transaction has a larger confidence its decendants are likely to collect chits in the future compared to others with lower confidence.
+For example T2 has larger confidence than T3 and its decendants are more likely to collect chits in the future.
+
+Similar to Bitcoin, Avalanche leaves determining the acceptance point of a transaction to the application.
+Committing a transaction can be performed through a safe early commitment.
+For virtuous transactions, T is accepted when it is the only transaction in its confliict set and has a confidence no less than a threshold of β1.
+If a virtuous transaction fails to get accepted due to a problem with parents, it could be accepted if reissued with different parents.
+
+### Transaction Entangling
+
+How Avalance entangles transactions:
+
+```text
+init:
+    `Set T` = Ø set of known transactions (empty set)
+    Q = Ø set of queried transactions (empty set)
+
+generateTx:
+    edges = {T' <— T: T' ∈ parentSelection(`Set T`)}
+    T = Tx(data, edges)
+    onReceiveEvent(T)
+
+onReceiveEvent(t)
+    if T ∉ `Set T` then
+        if PT (conflict set of T) = Ø then
+            PT = {T}, Pt.pref = T
+            PT.last = T, PT.cnt = 0
+        else PT = PT U {T}
+        `Set T` = T U {T}, cT = 0
+```
+
+Because transactions that generate and consume the same UTXO do not conflict with each other, any transaction can be reissued with different parents.
+
+### Node Main Loop
+
+Protocol main loop executed by each node:
+
+```text
+AvalancheLoop:
+    while true do
+        find T that satisfies T ∈ `Set T` AND T ∉ Q
+        K = sample(N\u, k)
+        P = SUM(v ∈ k) query(v, T)
+
+        if P ≥ α then
+            cT = 1 (chit for transaction T)
+            update preferences for ancestors:
+            for T' ∈ `Set T`: T' *<— T do
+                if T' confidence is larger than the confidence of preferred transation T` in the P conflict set - update the prefered transaction of the set:
+                if d(T') > d(PT'.pref) then
+                    PT'.pref = T'
+                if T' is not the last transaction in the T' conflict set - update the last place with it
+                if T' ≠ PT'.last then
+                    PT'.last = T', PT'.cnt=1
+                else
+                    ++PT'.cnt
+        else
+            for T' ∈ `Set T`: T' *<— T do
+                PT'.cnt = 0
+
+            otherwise confidence cnt remains 0 forever
+            Q = Q U {T} - mark T as queried
+```
+
+In each iteration the node tries to select a transaction that has not yet been queried.
+If no such transaction exists, the loop will stall until a new transation is added to the Set T.
+The node then select k number of peers and queries them.
+If more than α of these peers return a positive response, the chit value is set to 1.
```

Comment:
```suggestion
If more than `α` of these peers return a positive response, the chit value is set to `1`.
```

Reply:
This is fixed here https://github.com/optakt/optakt.github.io/pull/13/commits/e0e27e6be8c0aef4153af5e2add9722380c23a59#diff-2d4f16e70781857aec6689a748ad155e45611b81f184aee076bb139890410b98R161
Thanks!